base/libgit2/commit.jl

```diff
@@ -50,8 +50,8 @@ function commit(repo::GitRepo, msg::AbstractString;
                 refname::AbstractString=Consts.HEAD_FILE,
                 author::Signature = Signature(repo),
                 committer::Signature = Signature(repo),
-                tree_id::Oid = Oid(),
-                parent_ids::Vector{Oid}=Oid[])
+                tree_id::AbstractGitHash = GitHash(),
+                parent_ids::Vector{AbstractGitHash}=AbstractGitHash[])
```

Comment:
well now it might be a `Vector{GitHash}` or `Vector{GitShortHash}` which are not subtypes of `Vector{AbstractGitHash}`